internal/store/postgres/resource/resource.go

```diff
@@ -65,7 +66,8 @@ func FromModelToResource(r *Resource) (*resource.Resource, error) {
 	output, err := resource.NewResource(r.FullName, r.Kind, store, tnnt, metadata, r.Spec)
 	if err == nil {
 		output = resource.FromExisting(output, resource.ReplaceStatus(resource.FromStringToStatus(r.Status)))
-		output.UpdateURN(r.URN)
+		urn, _ := lib.ParseURN(r.URN)
```

Comment:
I have no opinion on this. The first thing in my mind is that, we have some assumptions (in other implementation as well) where we consider the ones written in DB as valid. But, at the same time, 3-4 lines above this consider otherwise. So, please suggest whether to handle it or ignore it.

Reply:
updated as discussed